### PR TITLE
bumping vpc cni version

### LINKED ISSUE #165: [FEATURE]upgrade vpc-cni on the EKS Module
**Is your feature request related to a problem? Please describe.**
`modules/compute/eks`

**Describe the solution you'd like**
The current `vpc-cni` version is getting outdated and should support the latest vpc-cni version to work with eks 1.29 version

### PR DESCRIPTION
**Pull request by srinivasreddych**
*Issue #, if available:*
Fixes #165 

*Description of changes:*
- upgraded `vpc-cni` version to `1.18.0` to support EKS 1.29 version
- This PR also fixed crashing `aws-node-agent` pod

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: 7d5a57669ff54125baeff8f3bb0fc957bb0f5f99
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=ee827b2c-3a35-4522-a343-7d4604fd1dc1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: 6c62986410b17d2537a1d7811c16f9385590d391
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=37a70623-afd9-462b-bfda-4376c9e46f77%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: 57970e50883c1b406dea00164baba4c93ffcfca7
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=cfea3a8d-a748-4e0e-8f58-00ede4f0a6fb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: 6678ce666b086adb0d975cbdc75e517be9a015b3
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=0d716c26-7e7c-4c0e-9a22-280ccce974cc%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: f009d163a86b7bb85a823db2af2fa2ad2d104a3b
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=449ddcb1-2169-4921-959e-c7c01ea96440%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: 3aadb18834c403c67e2ac66f1c0a5d7e34bae0e9
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=b9b571e0-2652-4698-9f5b-4c88ef80ed7f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: b9416ad1cba8b557e14b9d513d0b791a010616c9
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=72cc11ba-b589-4b38-bce3-2855959e96a6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: 3470b55764d3b70d811f477841c65b0589264908
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=2b14d91f-c2e2-4bfe-91fb-fcaf5141e6ed%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/compute/eks/stack.py [hunk 1]
```diff
@@ -319,6 +342,12 @@ def __init__(
                         },
                         "env": {"DISABLE_TCP_EARLY_DEMUX": True},
                     },
+                    "nodeAgent": {
+                        "image": {
+                            "region": self.region,
+                            "account": "602401143452",
```
Comment (kukushking) [inline modules/compute/eks/stack.py:348]: nit - should this be sanitised?

Comment (srinivasreddych) [reply modules/compute/eks/stack.py:348]: Did you mean the account number? This is a standard EKS Control plane team's account from which the images are pulled.

Comment (kukushking) [reply modules/compute/eks/stack.py:348]: Ah sorry thanks

**Review by kukushking — DISMISSED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: 08686a8002f276de4ba8f779d5e2eb06fd799b79
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=226a0819-6876-46ed-ad9e-bed94e3119a1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by malachi-constant — APPROVED** [review]

**Pull request merged**